### PR TITLE
Feature/support storage reset

### PR DESCRIPTION
**Pull request by RobertDeRose**
In support of PR https://github.com/alisaifee/flask-limiter/pull/49

### CODE REVIEW

**Comment by alisaifee** [comment]
Thanks @RobertDeRose. Just to understand the feature - what is the use case for having the `reset`

File: limits/storage.py [hunk 1]
```diff
@@ -221,6 +228,14 @@ def check(self):
         """
         return True
 
+    def reset(self):
+        self.timer.cancel()
+        self.storage = Counter()
```
Comment (alisaifee) [inline limits/storage.py]: Instead of creating new counters/dictionaries could you call `clear` on all of them?

**Comment by RobertDeRose** [comment]
So the use case for me, that I thought could be helpful for others as well, is that I use a lot of different limits with different keys, I wanted to be able to test that each limit was triggered in the expected way.
This was the main use case, for testing, right now I hack into the limit that Flask-Limiter uses and create a new MemoryStorage object and reset the weak reference. 

There might have been better options for this, I didn't know about hiro until I looked at your tests, but it still didn't fit my use case perfectly, I could have made it work probably, but this seemed a lot easier and I thought it could be useful to be able to programmatically reset your limits. I was thinking it wouldn't be too hard to add support redis, or memcached as long as you can match a prefix for the keys.

File: limits/storage.py (hunk 1 above)
Comment (RobertDeRose) [inline limits/storage.py]: I can do that.

**Comment by alisaifee** [comment]
Thanks for the explanation @RobertDeRose. I agree it might come in handy for someone else in their tests.

**Comment by RobertDeRose** [comment]
:tada: awesome man!

**Pull request merged**